src/application/service/CreateOrUpdateTest.ts

```diff
@@ -1,33 +1,53 @@
 import { UserId } from '../../domain/model/user/UserId';
 import { Test } from '../../domain/model/test/Test';
 import { TestId } from '../../domain/model/test/TestId';
+import { Results } from '../../domain/model/test/Results';
 import { TestRepository } from '../../domain/model/test/TestRepository';
 import { getTestTypes } from './index';
 import { TestTypeId } from '../../domain/model/testType/TestTypeId';
-import { Test as ApiTest } from '../../api/interface/index';
+import { TestCommand } from '../../api/interface/index';
+import jsonschema from 'jsonschema';
```

Comment:
Given we already have AJV which we need to validate the meta-schema couldn't we also use it to validate this schema? From what I saw 'jsonschema' doesn't allow meta-schema validation

Reply:
I didn't realise we had something that could do it.  Harsh was importing this one in package.json already.  Although I think he was only doing it with this use case in mind.

Reply:
I'll switch.  Simpler API as well.